packages/s3-store/index.ts

```diff
@@ -233,6 +278,61 @@ export class S3Store extends DataStore {
     return data.ETag as string
   }
 
+  private async downloadIncompletePart(id: string) {
+    const incompletePart = await this.getIncompletePart(id)
+
+    if (!incompletePart) {
+      return
+    }
+    const filePath = await this.uniqueTmpFileName('tus-s3-incomplete-part-')
+
+    try {
+      let incompletePartSize = 0
+
+      const byteCounterTransform = new stream.Transform({
+        transform(chunk, _, callback) {
+          incompletePartSize += chunk.length
+          callback(null, chunk)
+        },
+      })
+
+      // write to temporary file
+      await streamProm.pipeline(
+        incompletePart,
+        byteCounterTransform,
+        fs.createWriteStream(filePath)
+      )
+
+      const createReadStream = (options: {cleanUpOnEnd: boolean}) => {
```

Comment:
`cleanUpOnEnd` seems to always be true. Do we need this option then?

Reply:
Yhea, this option is more for awareness, that at the end of the stream, the file is removed.

It could hide the detail in which the file is being removed to the next developer looking at the code if we simply call: `createReadStream()` without the option.

Happy to remove it, but it seems clearer with it